lib/brainstem/dsl/configuration.rb

```diff
@@ -11,8 +11,13 @@ class Configuration
       # @params [Object] parent_configuration The parent configuration object
       #   which the new configuration object should use as a base.
       def initialize(parent_configuration = nil)
-        @parent_configuration = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
-        @storage = ActiveSupport::HashWithIndifferentAccess.new
+        @parent_configuration  = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
+        @storage               = ActiveSupport::HashWithIndifferentAccess.new
+
+        self.nonheritable_keys =
+          parent_configuration.respond_to?(:nonheritable_keys) &&
+          parent_configuration.nonheritable_keys ||
```

Comment:
That makes sense.  My question was about if this works:

```ruby
parent = Parent.new
child = Child.new
child.nonheritable!(:a)
parent.nonheritable!(:b)
```

what does `child.nonheritable_keys` print now?  With the array handling in the configuration, it would be an append only set and would include both, but your code might only have the child's?